src/main/java/tests/ShopServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package tests;
+
+import core.basesyntax.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ShopServiceTest {
+
+    private ShopService shopService;
+    private List<FruitTransaction> transactions;
+
+    @BeforeEach
+    public void setUp() {
```

Comment:
The setup method can be optimized by extracting the repeated creation of `FruitTransaction` objects into a separate method to avoid duplication in each test case.